docs/site/Extension-point-and-extensions.md

```diff
@@ -58,7 +63,47 @@ decorators and functions are provided to ensure consistency and convention.
 - `addExtension`: registers an extension class to the context for the named
   extension point
 
-The usage of these helper decorators and functions are illustrated in the
+## Examples
+
+1. Inject a getter function for extensions
+
+   ```ts
+   import {Getter} from '@loopback/context';
+   import {extensionPoint, extensions} from '@loopback/core';
+
+   @extensionPoint('greeters')
+   class GreetingService {
+     @extensions()
+     public getGreeters: Getter<Greeter[]>;
```

Comment:
If I have the traditional way of obtaining all the extensions using getGreeters(), why do we need
@extensions.view() or @extensions.list()? What is the advantage of having a context view compared to not having a context view? Or the list instead of the view.

The developer can call:

const greeters = await greeterService.getGreeters();
OR
const greeters = await greeterService.greetersView.values();
OR
const greeters = greeterService.greeters;

Sometimes too many choices is confusing, so a clear explanation for the purpose of each is important.

Reply:
The difference is not at fetching the list of greeters.

- getter: picks up latest list of greeters
- view: gives more control to handle fine-grained update of greeters (in addition to getter). I'm adding this feature to support a use case in extension point/extension pattern to allow fine-grained handling of an extension being removed. For example, if a cron job is removed after the app starts, we can stop the resolved instance if it exists.
- list: only contains a snapshot of greeters (no updates will be picked up. some extension point might want to have this behavior).

Reply:
@raymondfeng , thanks for the clarifications. This makes sense now. :)

Reply:
@raymondfeng , perhaps `beef up` the docs with these examples to help clarify the choices. thx.